File: src/test/java/fr/spoutnik87/musicbot_rest/AuthenticationHelperTest.java
package fr.spoutnik87.musicbot_rest;

import fr.spoutnik87.musicbot_rest.util.AuthenticationHelper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.test.context.support.WithUserDetails;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

@ExtendWith(SpringExtension.class)
@ContextConfiguration(classes = {BCryptTestConfig.class, SpringSecurityTestConfig.class})
public class AuthenticationHelperTest {

  @Test
  public void isAuthenticatedWhenUserNotAuthenticated() {
    assertFalse(AuthenticationHelper.isAuthenticated());
  }

  @Test
  @WithUserDetails("user@test.com")
  public void isAuthenticatedWhenUserAuthenticated() {
    assertTrue(AuthenticationHelper.isAuthenticated());
  }

  @Test
  @WithUserDetails("user@test.com")
  public void getAuthenticatedUserDetails() {
    UserDetails userDetails = AuthenticationHelper.getAuthenticatedUserDetails();
    assertEquals("user@test.com", userDetails.getUsername());
    assertNotNull(userDetails.getPassword());
    assertNotNull(userDetails.getAuthorities());
    assertTrue(
        ((SimpleGrantedAuthority) (userDetails.getAuthorities().toArray()[0]))
            .getAuthority()
            .equals("USER"));
  }

  @Test
  @WithUserDetails("user@test.com")
  public void getAuthenticatedUserEmail() {
    assertEquals("user@test.com", AuthenticationHelper.getAuthenticatedUserEmail());
  }

  @Test
  @WithUserDetails("user@test.com")
  public void getAuthenticatedUserAuthorities() {
    List<SimpleGrantedAuthority> simpleGrantedAuthorities =
        AuthenticationHelper.getAuthenticatedUserAuthorities();
    assertNotNull(simpleGrantedAuthorities);
    assertEquals("USER", simpleGrantedAuthorities.get(0).getAuthority());
  }

  @Test
  @WithUserDetails("user@test.com")
  public void isAuthenticatedUserInRole() {
    assertTrue(AuthenticationHelper.isAuthenticatedUserInRole("USER"));
    assertFalse(AuthenticationHelper.isAuthenticatedUserInRole("ADMIN"));
  }
}

File: src/test/kotlin/fr/spoutnik87/musicbot_rest/AuthenticationHelperTest.kt
package fr.spoutnik87.musicbot_rest

import fr.spoutnik87.musicbot_rest.constant.RoleEnum
import fr.spoutnik87.musicbot_rest.repository.UserRepository
import fr.spoutnik87.musicbot_rest.util.AuthenticationHelper
import fr.spoutnik87.musicbot_rest.util.WebSecurityTestConfig
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.mock.mockito.MockBean
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.test.context.support.WithMockUser
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [BCryptPasswordEncoder::class, WebSecurityTestConfig::class])
class AuthenticationHelperTest {

    @MockBean
    private lateinit var userRepository: UserRepository

    @Autowired
    private lateinit var bCryptPasswordEncoder: BCryptPasswordEncoder

    @Test
    fun isAuthenticated_NotAuthenticated_ReturnFalse() {
        assertFalse(AuthenticationHelper.isAuthenticated())
    }

    @Test
    @WithMockUser(username = "user@test.com", authorities = ["USER"])
    fun isAuthenticated_Authenticated_ReturnTrue() {
        assertTrue(AuthenticationHelper.isAuthenticated())
    }

    @Test
    @WithMockUser(username = "user@test.com", authorities = ["USER"])
    fun getAuthenticatedUserDetails_Authenticated_ReturnUserDetails() {
        val userDetails = AuthenticationHelper.getAuthenticatedUserDetails()
        assertEquals("user@test.com", userDetails?.username)
        assertNotNull(userDetails?.password)
        assertNotNull(userDetails?.authorities)
        assertEquals(RoleEnum.USER.value,
                (userDetails?.authorities!!.toTypedArray()[0] as SimpleGrantedAuthority).authority)
    }

    @Test
    @WithMockUser(username = "user@test.com", authorities = ["USER"])
    fun getAuthenticatedUserEmail_Authenticated_ReturnUserEmail() {
        assertEquals("user@test.com", AuthenticationHelper.getAuthenticatedUserEmail())
    }

    @Test
    @WithMockUser(username = "user@test.com", authorities = ["USER"])
    fun getAuthenticatedUserAuthorities_Authenticated_ReturnUserAuthorities() {
        val simpleGrantedAuthorities = AuthenticationHelper.getAuthenticatedUserAuthorities()
        assertNotNull(simpleGrantedAuthorities)
        assertEquals(RoleEnum.USER.value, simpleGrantedAuthorities?.get(0)?.authority)
    }

    @Test
    @WithMockUser(username = "user@test.com", authorities = ["USER"])
    fun isAuthenticatedUserInRole_AuthenticatedUser_ReturnTrueIfUser() {
        assertTrue(AuthenticationHelper.isAuthenticatedUserInRole(RoleEnum.USER))
        assertFalse(AuthenticationHelper.isAuthenticatedUserInRole(RoleEnum.ADMIN))
    }
}